Dockerfiles/agent/entrypoint/10-dogstatsd-socket.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Delete the dogstatsd unix socket if present
+# FIXME: move that logic to dsd itself
+
+if [[ -z "${DD_DOGSTATSD_SOCKET}" ]]; then
+    export DD_DOGSTATSD_NON_LOCAL_TRAFFIC=1
```

Comment:
In **50-ecs.sh**, **50-kubernetes.sh** you are using some exit-fast mechanism and it's pretty cool to read.

Is it possible to do it here ?
Like:
```bash
if [[ -z "${DD_DOGSTATSD_SOCKET}" ]]; then
    export DD_DOGSTATSD_NON_LOCAL_TRAFFIC=1
    exit 0
fi
```
Just to be sure, is this file sourced ?

By adding `set -e` we also can avoid the `|| exit $?`